tables/TaQL/ExprNodeSet.cc

```diff
@@ -244,14 +244,15 @@ TENShPtr TableExprNodeSet::setOrArray() const
         // Set ndim and shape if those are constant.
     }
     TableExprNodeSet* set = new TableExprNodeSet (*this);
+    TENShPtr pset(set);
     if (itsBounded) {
         // Set the type to VTArray; the getArray functions
         // will convert the set to an array for each row.
         // Set the shape and/or dimensionaly if constant.
         set->setValueType (VTArray);
         set->setShape();
     }
-    return set;
+    return pset;
```

Comment:
When updating line 246, you can revert this change.
```suggestion
    return set;
```

Reply:
done